netmiko/fortinet/fortinet_ssh.py

```diff
@@ -1,104 +1,298 @@
-        """Not Implemented"""
-        raise NotImplementedError
+from netmiko.base_connection import BaseConnection
+from netmiko import log
+from netmiko.exceptions import (
+	NetmikoTimeoutException,
+	NetmikoAuthenticationException,
+	ConfigInvalidException,
+)
+
+
+class FortinetSSH(BaseConnection):
+	def _modify_connection_params(self) -> None:
+		"""Modify connection parameters prior to SSH connection."""
+		paramiko_transport = getattr(paramiko, "Transport")
+		paramiko_transport._preferred_kex = (
+			"diffie-hellman-group14-sha1",
+			"diffie-hellman-group-exchange-sha1",
+			"diffie-hellman-group-exchange-sha256",
+			"diffie-hellman-group1-sha1",
+		)
+
+	def session_preparation(self) -> None:
+		"""Prepare the session after the connection has been established."""
+
+		data = self._test_channel_read(pattern="to accept|[#$]")
+		# If "set post-login-banner enable" is set it will require you to press 'a'
+		# to accept the banner before you login. This will accept if it occurs
+		if "to accept" in data:
+			self.write_channel("a\r")
+			self._test_channel_read(pattern=r"[#$]")
+
+		self.set_base_prompt(alt_prompt_terminator="$")
+		self.disable_paging()
+		self.clear_buffer()
+
+	def set_base_prompt(self, 
+		pri_prompt_terminator="#", 
+		alt_prompt_terminator="$", 
+		delay_factor=1.0,
+		pattern = r"(#|\$)"
+		):
+		"""Sets self.base_prompt
+
+		Used as delimiter for stripping of trailing prompt in output.
+
+		Should be set to something that is general and applies in multiple contexts. For Cisco
+		devices this will be set to router hostname (i.e. prompt without > or #).
+
+		This will be set on entering user exec or privileged exec on Cisco, but not when
+		entering/exiting config mode.
+
+		:param pri_prompt_terminator: Primary trailing delimiter for identifying a device prompt
+
+		:param alt_prompt_terminator: Alternate trailing delimiter for identifying a device prompt
+
+		:param delay_factor: See __init__: global_delay_factor
+
+		:param pattern: Regular expression pattern to search for in find_prompt() call
+		"""
+		
+		return super().set_base_prompt(
+			pri_prompt_terminator=pri_prompt_terminator, 
+			alt_prompt_terminator=alt_prompt_terminator,
+			delay_factor=delay_factor,
+			pattern = pattern
+		)
+
+	def find_prompt(
+		self, 
+		delay_factor: float = 1.0, 
+		pattern: str = r"(#|\$)", 
+		) -> str:
+		"""Finds the current network device prompt, last line only.
+
+		:param delay_factor: See __init__: global_delay_factor
+		:type delay_factor: int
+
+		:param pattern: Regular expression pattern to determine whether prompt is valid
+
+		:param confirm_pattern: Regular expression pattern to confirm prompt was found by auto discovery.
+		"""
+		return super().find_prompt(
+			delay_factor=delay_factor, 
+			pattern=pattern,
+		)
+
+	def send_config_set(
+		self,
+		config_commands = None,
+		exit_config_mode: bool = True,
+		read_timeout: float = None,
+		delay_factor: float = 1.0,
+		max_loops: int = 150,
```

Comment:
`delay_factor` and `max_loops` should both default to None. They are largely deprecated.

Reply:
Done